src/error.rs

```diff
@@ -139,6 +156,18 @@ impl de::Error for Error {
         })
     }
 
+    #[cfg(not(feature = "std"))]
+    fn custom<T>(_msg: T) -> Error
+    where
+        T: fmt::Display,
+    {
+        // TODO propagate at least something
```

Comment:
What should be done here?

Reply:
Right now we just set the code to `ErrorCode::Custom` without any extra data on what caused the error. Ideally I would like to propagate the error message here, but that would only be possible if we could Box up the message. The next best thing would be to propagate part of the error, but I have no good ideas for how to do that at the moment.